src/snowflake/cli/api/entities/snowpark_entity.py

```diff
@@ -1,19 +1,27 @@
+from typing import Generic, TypeVar
+
 from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli.api.project.schemas.entities.snowpark_entity import (
+    FunctionEntityModel,
+    ProcedureEntityModel,
+)
+
+T = TypeVar("T")
 
 
-class SnowparkEntity(EntityBase):
+class SnowparkEntity(EntityBase[Generic[T]]):
     pass
 
 
-class FunctionEntity(SnowparkEntity):
+class FunctionEntity(SnowparkEntity[FunctionEntityModel]):
     """
     A single UDF
```

Comment:
UDF is not the only kind of function in Snowpark (they have UDTF and UDAF as well)

Reply:
I'll let @sfc-gh-jsikorski and @sfc-gh-turbaszek address this in a future PR